diff --git a/elf.c b/elf.c
index 19b7960..5590b27 100644
--- a/elf.c
+++ b/elf.c
@@ -550,7 +550,7 @@ elf_crc32_file (struct backtrace_state *state, int descriptor,
 /* A dummy callback function used when we can't find a symbol
    table.  */
 
-static void
+void
 elf_nosyms (struct backtrace_state *state ATTRIBUTE_UNUSED,
 	    uintptr_t addr ATTRIBUTE_UNUSED,
 	    backtrace_syminfo_callback callback ATTRIBUTE_UNUSED,
@@ -751,7 +751,7 @@ elf_add_syminfo_data (struct backtrace_state *state,
 
 /* Return the symbol name and value for an ADDR.  */
 
-static void
+void
 elf_syminfo (struct backtrace_state *state, uintptr_t addr,
 	     backtrace_syminfo_callback callback,
 	     backtrace_error_callback error_callback ATTRIBUTE_UNUSED,
@@ -3981,7 +3981,7 @@ backtrace_uncompress_lzma (struct backtrace_state *state,
    elf_add will need to be called on the descriptor again after
    base_address is determined.  */
 
-static int
+int
 elf_add (struct backtrace_state *state, const char *filename, int descriptor,
 	 const unsigned char *memory, size_t memory_size,
 	 uintptr_t base_address, backtrace_error_callback error_callback,
diff --git a/fileline.c b/fileline.c
index 0472f47..8d22e07 100644
--- a/fileline.c
+++ b/fileline.c
@@ -158,7 +158,7 @@ macho_get_executable_path (struct backtrace_state *state,
 /* Initialize the fileline information from the executable.  Returns 1
    on success, 0 on failure.  */
 
-static int
+int
 fileline_initialize (struct backtrace_state *state,
 		     backtrace_error_callback error_callback, void *data)
 {
diff --git a/internal.h b/internal.h
index bb481f3..a1b7ec9 100644
--- a/internal.h
+++ b/internal.h
@@ -377,4 +377,24 @@ extern int backtrace_uncompress_lzma (struct backtrace_state *,
 				      unsigned char **uncompressed,
 				      size_t *uncompressed_size);
 
+extern int elf_add (struct backtrace_state *state, const char *filename, int descriptor,
+                    const unsigned char *memory, size_t memory_size,
+                    uintptr_t base_address, backtrace_error_callback error_callback,
+                    void *data, fileline *fileline_fn, int *found_sym, int *found_dwarf,
+                    struct dwarf_data **fileline_entry, int exe, int debuginfo,
+                    const char *with_buildid_data, uint32_t with_buildid_size);
+
+extern void elf_syminfo (struct backtrace_state *state, uintptr_t addr,
+                         backtrace_syminfo_callback callback,
+                         backtrace_error_callback error_callback,
+                         void *data);
+
+extern void elf_nosyms (struct backtrace_state *state,
+                        uintptr_t addr,
+                        backtrace_syminfo_callback callback,
+                        backtrace_error_callback error_callback, void *data);
+
+extern int fileline_initialize (struct backtrace_state *state,
+                                backtrace_error_callback error_callback, void *data);
+
 #endif
